controllers/oas_parser.go

```diff
@@ -0,0 +1,41 @@
+package controllers
+
+import (
+	"encoding/json"
+
+	"github.com/buger/jsonparser"
+)
+
+var ClientCertificatePath = []string{TykOASExtenstionStr, "server", "clientCertificates", "allowlist"}
+
+func OASSetClientCertificates(data []byte, certid string) ([]byte, error) {
+	allowlist := []string{}
+
+	val, _, _, err := jsonparser.Get(data, ClientCertificatePath...)
+	if err != nil && err != jsonparser.KeyPathNotFoundError {
+		return nil, err
+	}
+
+	if val != nil {
+		_, err = jsonparser.ArrayEach(val, func(value []byte, dataType jsonparser.ValueType, offset int, err error) {
+			item, pErr := jsonparser.ParseString(value)
+			if pErr != nil {
+				return
```

Comment:
if `pErr` is not nil, we return `err` not `pErr`. Is this behaviour expected? I assume we need to return pErr unless I miss some specific implementation details. Here, do we want to continue parsing certificates array if the parser is not able to parse the field within the array?

Reply:
The function `ArrayEach` accepts doesn't return an error. So I can't return pErr from it.
I am checking error returned by ArrayEach itself.

Reply:
oh my bad, you are right